pkg/planner/core/planbuilder.go

```diff
@@ -78,10 +79,21 @@ type visitInfo struct {
 	column           string
 	err              error
 	alterWritable    bool
-	dynamicPriv      string
+	dynamicPrivs     []string
 	dynamicWithGrant bool
 }
 
+func (v *visitInfo) Equals(other *visitInfo) bool {
+	return v.privilege == other.privilege &&
+		v.db == other.db &&
+		v.table == other.table &&
+		v.column == other.column &&
+		v.err == other.err &&
+		v.alterWritable == other.alterWritable &&
+		reflect.DeepEqual(v.dynamicPrivs, other.dynamicPrivs) &&
```

Comment:
If this function is only for testing, DeepEqual can be directly used in the test.

Reply:
Can't directly use DeepEqual because `v.err` is `error` type and should not use DeepEqual for error.